src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -61,59 +64,112 @@ const DeleteWorkspaceModal = ({ workspace, workspace: { workspace: { name, bucke
+        icon('warning-standard', { size: 24, color: colors.warning() }),
+        span({ style: { marginLeft: '1ch' } }, ['Delete workspace'])
+      ]),
+      onDismiss,
+      okButton: h(ButtonPrimary, {
+        disabled: _.toLower(deleteConfirmation) !== 'delete workspace' || isDeleteDisabledFromResources,
+        onClick: deleteWorkspace,
+        tooltip: Utils.cond(
+          [isDeleteDisabledFromResources && isGoogleWorkspace(workspace), () => 'You must ensure all apps in this workspace are deletable'],
+          [isDeleteDisabledFromResources && isAzureWorkspace(workspace), () => 'This workspace cannot be deleted'],
+          [_.toLower(deleteConfirmation) !== 'delete workspace', () => 'You must type the confirmation message'],
+          () => '')
+      }, 'Delete workspace'),
+      styles: { modal: { background: colors.warning(0.1) } }
+    }, [
+      div(['Are you sure you want to permanently delete the workspace ',
+        span({ style: { fontWeight: 600, wordBreak: 'break-word' } }, name),
+        '?']),
+      getStorageDeletionMessage(),
+      isDeleteDisabledFromResources && div({ style: { marginTop: '1rem' } }, [
+        getResourceDeletionMessage()
+      ]),
+      !isDeleteDisabledFromResources && hasApps() && div({ style: { marginTop: '1rem' } }, [
+        p(['Deleting it will also delete any associated applications:']),
+        getResourceDeletionMessage()
+      ]),
+      collaboratorEmails && collaboratorEmails.length > 0 && div({ style: { marginTop: '1rem' } }, [
+        p(`${pluralize('collaborator', collaboratorEmails.length, true)} will lose access to this workspace.`),
+        div(collaboratorEmails.slice(0, 5).map(
+          email => div({ key: email, style: { overflow: 'hidden', textOverflow: 'ellipsis' } }, [h(Link, { href: `mailto:${email}` }, [email])])
+        )),
+        collaboratorEmails.length > 5 && (
+          div(`and ${collaboratorEmails.length - 5} more`)
+        )
+      ]),
+      !isDeleteDisabledFromResources && b({ style: { display: 'block', marginTop: '1rem' } }, 'This cannot be undone.'),
+      !isDeleteDisabledFromResources && div({ style: { display: 'flex', flexDirection: 'column', marginTop: '1rem' } }, [
+        label({ htmlFor: 'delete-workspace-confirmation', style: { marginBottom: '0.25rem' } }, ['Please type \'Delete Workspace\' to continue:']),
+        h(TextInput, {
+          id: 'delete-workspace-confirmation',
+          placeholder: 'Delete Workspace',
+          value: deleteConfirmation,
+          onChange: setDeleteConfirmation
+        })
+      ]),
+      (deleting || loading) && spinnerOverlay
+    ])
+  }
+
+  if (loading) {
+    // only show the deletion modal when we have all of our constiuent data elements
+    return div([
+      spinnerOverlay
+    ])
+  } else if (isAzureWorkspace(workspace) && (hasApps() || hasRuntimes() || deletingResources)) {
+    return getAzureResourceCleanupModal()
```

Comment:
If they're left as render functions, I'd vote for changing the name to "render...".

But it looks like there's a lot going on in this function, so it would be nice to split it up into smaller, focused, components.

Reply:
I've renamed these functions with `render` prefixes -- agreed they should likely be refactored but I'd like to keep the scope of this PR constrained. @cahrens @nawatts let me know what you think.